File: frontend/src/services/taskService.ts
import axios from 'axios';
import type { Task, TaskRequest } from '../types/task';

const API_BASE_URL = 'http://localhost:8080/api';

const apiClient = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

export const taskService = {
  getAllTasks: async (completed?: boolean): Promise<Task[]> => {
    const params = completed !== undefined ? { completed } : {};
    const response = await apiClient.get<Task[]>('/tasks', { params });
    return response.data;
  },

  getTaskById: async (id: number): Promise<Task> => {
    const response = await apiClient.get<Task>(`/tasks/${id}`);
    return response.data;
  },

  createTask: async (task: TaskRequest): Promise<Task> => {
    const response = await apiClient.post<Task>('/tasks', task);
    return response.data;
  },

  updateTask: async (id: number, task: TaskRequest): Promise<Task> => {
    const response = await apiClient.put<Task>(`/tasks/${id}`, task);
    return response.data;
  },

  deleteTask: async (id: number): Promise<void> => {
    await apiClient.delete(`/tasks/${id}`);
  },
};

File: backend/src/main/java/com/todolist/service/TaskService.java
package com.todolist.service;

import com.todolist.dto.TaskRequest;
import com.todolist.dto.TaskResponse;

import java.util.List;
import java.util.Optional;

/**
 * 任务服务接口
 */
public interface TaskService {
    
    /**
     * 获取所有任务，支持按完成状态过滤
     *
     * @param completed 任务完成状态（可选）
     * @return 任务响应列表
     */
    List<TaskResponse> getAllTasks(Boolean completed);
    
    /**
     * 根据ID获取任务
     *
     * @param id 任务ID
     * @return 任务响应（如果存在）
     */
    Optional<TaskResponse> getTaskById(Long id);
    
    /**
     * 创建新任务
     *
     * @param request 任务请求体
     * @return 创建的任务响应
     */
    TaskResponse createTask(TaskRequest request);
    
    /**
     * 更新任务
     *
     * @param id 任务ID
     * @param request 任务请求体
     * @return 更新后的任务响应（如果存在）
     */
    Optional<TaskResponse> updateTask(Long id, TaskRequest request);
    
    /**
     * 删除任务
     *
     * @param id 任务ID
     * @return 是否删除成功
     */
    boolean deleteTask(Long id);
}
